main.go

```diff
@@ -1,3 +1,20 @@
 package main
 
-func main() {}
+import (
+	"flag"
+	"fmt"
+
+	"github.com/gin-gonic/gin"
+	"github.com/mhghw/fara-message/api"
+)
+
+var port = flag.Int("port", 8080, "Port to run the HTTP server")
+
+func main() {
+	flag.Parse()
+	addr := fmt.Sprintf(":%d", *port)
+	router := gin.Default()
+	router.POST("/user/register", api.Register)
```

Comment:
everyting related to web server should be in api package, not here!
create a web server in api that gets a port, then do all what you did here (except parsing the flag) in api package.